circuit_maintenance_parser/parsers/zayo.py

```diff
@@ -32,10 +42,23 @@ def parse_bs(self, btags: ResultSet, data: dict):
             if isinstance(line, bs4.element.Tag):
                 if line.text.lower().strip().startswith("maintenance ticket #:"):
                     data["maintenance_id"] = self.clean_line(line.next_sibling)
-                elif line.text.lower().strip().startswith("urgency:"):
-                    urgency = self.clean_line(line.next_sibling)
-                    if urgency == "Planned":
+                elif "serves as official notification" in line.text.lower():
+                    if "will be performing maintenance" in line.text.lower():
                         data["status"] = Status("CONFIRMED")
+                    elif "has cancelled" in line.text.lower():
+                        data["status"] = Status("CANCELLED")
+                # TODO: some Zayo notifications may include multiple activity dates. How best to handle this?
```

Comment:
I'm guessing this may be one of the first instances where one email could theoretically generate multiple maintenances?  I'm guessing it would be a big change to change that cardinality all the way through...

I wonder if we could leverage the tagging feature in the other open PR to say that an email needs to get re-processed for a 2nd/3rd/nth date?  That could also be tricky though in terms of tracking the state of which activity needs to get re-processed.

Reply:
The constraint I see here is that the Maintenance ID is unique for the 3 periods, so I see it as one Maintenance with a non continuous time window. It's not an easy problem to solve because it is not matching the "[standard](https://github.com/jda/maintnote-std/blob/master/standard.md)" that this library is following. More specific, in this [section](https://github.com/jda/maintnote-std/blob/master/standard.md#35-event-component-and-associated-properties) the DTSTART and DTEND are unique for each notification.

Brainstorming some options:

- Get the overall maintenance time, taking the first START date and the last END date. From the example: `01-Nov-2021 06:01 - 03-Nov-2021 11:00`. This is the easiest one to implement and the issue is that there is lost of information for the time slots not affected.
- Create different Maintenances from the original one with different Maintenance IDs adding some suffix or prefix, for example from TTN-0001 we get TTN-0001-a, TTN-0001-b, TTN-0001-c. This is still keeping the standard like the one above (the parser supports taking multiple maintenances from a notification), but the client could get some misunderstanding from this.
- Create different Maintenances from the original one with same Maintenance ID, but with some difference, for instance using the `UID` filter to differentiate them, so each one would have UID=1, UID=2 and UID=3. This could mean some changes on the parser clients if they want to handle them as unique entities.

Open to feedback

Reply:
I'm thinking that the first option (treat it as one long extended maintenance) is probably the simplest to implement and is the most conservative option in terms of blocking an overall window where the circuit may be affected.